### PR TITLE
Action !takedown labels in hydrator

### PR DESCRIPTION
**Pull request by dholms**
Apply takedowns to server responses based on if some content has a `!takedown` or `!suspend` label from a configured labeler

### CODE REVIEW

File: packages/bsky/src/hydration/hydrator.ts [hunk 1]
```diff
@@ -724,3 +755,23 @@ const mergeManyStates = (...states: HydrationState[]) => {
...
+    const subjectLabels = labels.get(key)
+    if (!subjectLabels) continue
+    if (includesTakedownLabel(subjectLabels)) {
+      hydrationMap.set(key, null)
+    }
+  }
+}
+
+const includesTakedownLabel = (labels: Label[]) => {
+  return labels.some((l) => l.val === TAKEDOWN_LABEL)
```
Comment (devinivy) [inline packages/bsky/src/hydration/hydrator.ts:774]: Is it guaranteed that none of the labels here have a `neg: true`?

**Review by devinivy — APPROVED** [review]
Sweet!

**Comment by bnewbold** [comment]
are we going to do the `!suspend` label at the same time? for time-limited. I guess we could have just the one label with/without time-limits, but it feels a bit useful to make it really explicit in both protocol and UI/UX whether the takedown is temporary or not.

File: packages/bsky/src/hydration/hydrator.ts [hunk 2]
```diff
@@ -724,3 +755,23 @@ const mergeManyStates = (...states: HydrationState[]) => {
...
+}
```
[the remark this answers is not in the document]
Comment (dholms) [reply packages/bsky/src/hydration/hydrator.ts]: yeah I like it - I added a materialized `isTakendown` flag to the label hydration map

Also makes it easier to extend the functionality with either more labels that toggle the flag or other types of flags 👍

File: packages/bsky/src/hydration/hydrator.ts (hunk 1 above)
Comment (dholms) [reply packages/bsky/src/hydration/hydrator.ts]: Should be, but I'll add a check just in case 👍

**Review by devinivy — APPROVED** [review]

File: packages/bsky/src/views/index.ts [hunk 3]
```diff
@@ -61,7 +61,9 @@ export class Views {
   // ------------
 
   actorIsTakendown(did: string, state: HydrationState): boolean {
-    return !!state.actors?.get(did)?.takedownRef
+    if (state.actors?.get(did)?.takedownRef) return true
+    if (state.labels?.get(did)?.isTakendown) return true
```
[the remark this answers is not in the document]
Comment (dholms) [reply packages/bsky/src/views/index.ts:65]: I think we can punt for now as we're still pairing those with pds takedowns

[the remark this answers is not in the document]
Comment (dholms) [reply packages/bsky/src/views/index.ts:65]: But agreed, needs to be addressed soon 👍

**Pull request merged**